### DISCUSSION TITLE
Many buckets or one - which is preferred?

### DISCUSSION THREAD

**Discussion by bruunand** [minio/minio#14665] [Q&A] [answered]
What should one consider when choosing between either
- having multiple buckets, or
- putting objects in one bucket and making use of prefixes to organize the objects? 

For example, a common strategy in Elasticsearch is to make daily buckets. Would it make sense to do something similar with MinIO, or does it make more sense from e.g. a performance standpoint to do `bucket/yy/mm/dd/object`?

**Comment by klauspost** [comment, marked as answer]
Yes (there is no "correct" answer)

Buckets are very top-level constructs. Use them, but use them sparingly. They can be useful for separating applications and data types that require different handling, for example with lifecycle, replication, etc.

Keep bucket count easy to manually manage. AWS enforces less than 100. MinIO isn't as strict, but it is a good target to keep in mind.
